sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -325,6 +325,16 @@ private static string JoinHeaderValues(HeaderStringValues values)
                 return string.Empty;
             }
 
+            // Special case when HeaderStringValues.Count == 1, because HttpHeaders also special cases it and creates HeaderStringValues instance from a single string
+            // https://github.com/dotnet/runtime/blob/ef5e27eacecf34a36d72a8feb9082f408779675a/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeadersNonValidated.cs#L150
+            // https://github.com/dotnet/runtime/blob/ef5e27eacecf34a36d72a8feb9082f408779675a/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L1105
+            // Which is later used in HeaderStringValues.ToString:
+            // https://github.com/dotnet/runtime/blob/729bf92e6e2f91aa337da9459bef079b14a0bf34/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HeaderStringValues.cs#L47
+            if (count == 1)
```

Comment:
Can we add a test?

Reply:
What kind of test?

Reply:
A test where there is a single header?

Reply:
Perhaps this is already covered, but it would be good to verify.

Reply:
It turns out that I have run benchmark over the wrong `string.Join` overload. `HeaderStringValues.ToString()` is better for most of the cases.

Reply:
Ok, now I recall why I have added this in the first place. `ToString` has different separator value. I'll leave perf test here and add a comment.

Reply:
`SettingContentStreamPreservesHeaders` definitely covers the `count == 1` case, but I don't see it covering the zero or >1 cases.

Reply:
>1 cases is covered in `CanGetAndSetMultiValueResponseHeaders` and `ResponseHeadersAreSplit`